app/views/hosts/_form.html.erb

```diff
@@ -121,7 +122,7 @@
           :help_inline => _("Include this host within Foreman reporting")
         %>
         <div id='model_name'>
-          <%= select_f f, :model_id, Model.all, :id, :to_label, { :include_blank => true }, {:label => _("Hardware Model")} unless @host.compute_resource_id%>
+          <%= select_f f, :model_id, Model.authorized(:view_models), :id, :to_label, { :include_blank => true }, {:label => _("Hardware Model")} unless @host.compute_resource_id%>
```

Comment:
`accessible_models` ? If it's in the helper, we can at least write a simple, cheap, unit test that expects `.authorized` on Model and "ensure" at least we notice if we break it.